src/admin_access_token.py

```diff
@@ -41,20 +42,36 @@ def __init__(self, app: ops.Application, model: ops.Model):
         self._model = model
 
     def get(self, container: ops.Container) -> typing.Optional[str]:
-        """Get admin access token.
+        """Get an admin access token.
+
+        If the admin token is not valid or it does not exist it creates one.
 
         Args:
             container: Workload container.
 
         Returns:
             admin access token or None if fails.
         """
+        admin_access_token = self._get_from_peer_relation()
+        if admin_access_token is None or not synapse.is_token_valid(admin_access_token):
+            admin_access_token = self._renew_token(container)
+        return admin_access_token
+
+    def _get_from_peer_relation(self) -> typing.Optional[str]:
+        """Get admin access token from the peer relation.
+
+        Returns:
+            Admin access token or None if admin token was not found or there was an error.
+
+        Raise:
+            PeerRelationNotFoundError: if the peer relation does not exist.
+        """
         peer_relation = self._model.get_relation(PEER_RELATION_NAME)
         if not peer_relation:
             logger.error(
                 "Failed to get admin access token: no peer relation %s found", PEER_RELATION_NAME
             )
-            return None
+            raise PeerRelationNotFoundError()
```

Comment:
Isn't line 75 unreachable now?

Reply:
it is reacheble (otherwise this function would always fail)

Reply:
Oops, my bad